mmdet/models/backbones/mobilenetv2.py

```diff
@@ -0,0 +1,253 @@
+def conv_bn(inp, oup, stride, groups=1, act_fn=nn.ReLU):
+    return nn.Sequential(
+        nn.Conv2d(inp, oup, 3, stride, 1, bias=False, groups=groups),
+        nn.BatchNorm2d(oup),
+        act_fn(inplace=True)
+    )
+
+
+def conv_1x1_bn(inp, oup, groups=1, act_fn=nn.ReLU):
+    return nn.Sequential(
+        nn.Conv2d(inp, oup, 1, 1, 0, bias=False, groups=groups),
+        nn.BatchNorm2d(oup),
+        act_fn(inplace=True)
+    )
+
+
+class InvertedResidual(nn.Module):
+    def __init__(self, inp, oup, stride, expand_ratio, ssd_output=False,
+                 activation=nn.ReLU):
+        super(InvertedResidual, self).__init__()
+        self.stride = stride
+        assert stride in [1, 2]
+
+        hidden_dim = int(inp * expand_ratio)
+        inp = int(inp)
+        oup = int(oup)
+        self.use_res_connect = self.stride == 1 and inp == oup
+
+        if expand_ratio == 1:
+            self.conv = nn.Sequential(
+                # dw
+                nn.Conv2d(hidden_dim, hidden_dim, 3, stride, 1,
+                          groups=hidden_dim, bias=False),
+                nn.BatchNorm2d(hidden_dim),
+                activation(inplace=True),
+                # pw-linear
+                nn.Conv2d(hidden_dim, oup, 1, 1, 0, bias=False),
+                nn.BatchNorm2d(oup),
+            )
+        else:
+            self.conv = [
+                # pw
+                nn.Conv2d(inp, hidden_dim, 1, 1, 0, bias=False),
+                nn.BatchNorm2d(hidden_dim),
+                activation(inplace=True),
+                # dw
+                nn.Conv2d(hidden_dim, hidden_dim, 3, stride, 1,
+                          groups=hidden_dim, bias=False),
+                nn.BatchNorm2d(hidden_dim),
+                activation(inplace=True),
+                # pw-linear
+                nn.Conv2d(hidden_dim, oup, 1, 1, 0, bias=False),
+                nn.BatchNorm2d(oup),
+            ]
+        self.ssd_output = ssd_output
+        if self.ssd_output:
+            self.output_conv = conv_1x1_bn(hidden_dim, hidden_dim, 1,
+                                           activation)
+            self.conv1 = nn.Sequential(*self.conv[0:3])
+            self.conv2 = nn.Sequential(*self.conv[3:])
+        else:
+            self.conv = nn.Sequential(*self.conv)
+
+    def compute_conv(self, x):
+        if not self.ssd_output:
+            return self.conv(x)
+
+        expanded_x = self.output_conv(self.conv1(x))
+        return (expanded_x, self.conv2(expanded_x))
+
+    def forward(self, x):
+        if self.use_res_connect:
+            if self.ssd_output:
+                expanded, conv = self.compute_conv(x)
+                return (expanded, x + conv)
+            return x + self.compute_conv(x)
+        else:
+            return self.compute_conv(x)
+
+
+@BACKBONES.register_module
+class SSDMobilenetV2(nn.Module):
+    def __init__(self, input_size, width_mult=1.0,
+                 activation_type='relu',
+                 single_scale=False):
```

Comment:
Similar to SSDVGG, we prefer all hyper-parameters can be defined with one dict. Therefore, it would be great if there are no hard code settings in the __init__().